kover-jvm-agent/src/main/java/kotlinx/kover/jvmagent/KoverJvmAgentPremain.java

```diff
@@ -0,0 +1,119 @@
+package kotlinx.kover.jvmagent;
+
+import java.io.*;
+import java.lang.instrument.Instrumentation;
+import java.util.HashSet;
+
+import static kotlinx.kover.jvmagent.ParseUtils.isBoolean;
+
+public class KoverJvmAgentPremain {
+
+    private static final String FILE_PREFIX_IN_ARGS = "file:";
+    private static final String FILE_PATH_ATTRIBUTE = "report.file=";
+    private static final String APPEND_ATTRIBUTE = "report.append=";
+    private static final String EXCLUDE_WITH_WILDCARDS_ATTRIBUTE = "exclude=";
+    private static final String EXCLUDE_WITH_REGEX_ATTRIBUTE = "exclude.regex=";
+    private static final String INCLUDE_WITH_WILDCARDS_ATTRIBUTE = "include=";
+    private static final String INCLUDE_WITH_REGEX_ATTRIBUTE = "include.regex=";
+
+    private static final String regexMetacharacters = "<([{\\^-=$!|]})+.>";
+
+    private static final HashSet<Character> regexMetacharactersSet = new HashSet<Character>();
+
+    static {
+        for (int i = 0; i < regexMetacharacters.length(); i++) {
+            char c = regexMetacharacters.charAt(i);
+            regexMetacharactersSet.add(c);
+        }
+    }
+
+    public static void premain(String argsString, Instrumentation instrumentation) throws Exception {
+        KoverAgentSettings settings = readSettingsFromFile(extractArgsFile(argsString));
+        IntellijIntegration.callPremain(settings, instrumentation);
+    }
+
+    private static File extractArgsFile(String args) {
+        if (!args.startsWith(FILE_PREFIX_IN_ARGS)) {
+            throw new IllegalArgumentException("Incorrect arguments format for Kover JVM agent, arguments are expected to start with " + FILE_PREFIX_IN_ARGS);
```

Comment:
Are you sure that allowing passing arguments only in file and not accepting them in CLI is a good idea? Maybe some users don't want them to be in file?

Reply:
In the future, we can add it, but so far there has not been such a request, and I do not want to come up with a format ahead of time